src/main/java/in/projecteka/consentmanager/user/OtpAttemptRepository.java

```diff
@@ -0,0 +1,81 @@
+package in.projecteka.consentmanager.user;
+
+import in.projecteka.consentmanager.common.DbOperationError;
+import in.projecteka.consentmanager.user.model.OtpAttempt;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.Tuple;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Repository
+@AllArgsConstructor
+public class OtpAttemptRepository {
+
+    private static final String INSERT_OTP_ATTEMPT = "INSERT INTO " +
+            "otp_attempt (session_id ,cm_id, identifier_type, identifier_value, status, action) VALUES ($1,$2,$3,$4,$5,$6)";
+
+    private static final String SELECT_OTP_ATTEMPT = "SELECT session_id,identifier_type,identifier_value,status,attempt_at,action,cm_id FROM otp_attempt " +
+            "WHERE identifier_value = $1 AND action = $3 AND cm_id = $4 AND identifier_type = $5" +
```

Comment:
is it possible to put the parameters in sequence?

Reply:
also any specific reason to limit the query to maxotpattemp?  the query has got enough where clause to look for specific action and  identifier and cm id. would it not suffice?

Reply:
There can be more records matching with the parameters. So we are limiting that to return on latest `maxOtpAttemts`.